sympy/sets/tests/test_sets.py

```diff
@@ -1597,6 +1597,16 @@ def test_issue_20089():
     assert A.issubset(C)
     assert B.issubset(C)
 
+def test_issue_19378():
+    a = FiniteSet(1, 2)
+    b = ProductSet(a, a)
+    c = FiniteSet((1, 1), (1, 2), (2, 1), (2, 2))
+    assert b.is_subset(c)
+    d = FiniteSet(1)
+    assert not b.is_subset(d)
+    assert Eq(b, c).simplify()
+    assert not Eq(a, c).simplify()
```

Comment:
Tests that use `assert not f()` or `assert f()` should not be used. Almost any object returned by `f()` can be truthy so we should be more precise e.g.:
```
    assert Eq(b, c).simplify() is S.true
```
Likewise `None` can be falsey and we should not return `None` from `Eq.simplify` but this PR does:
```python
In [5]: x = Symbol('x')                                                                                                           

In [6]: e = Eq({1}, {x})                                                                                                          

In [7]: e                                                                                                                         
Out[7]: {1} = {x}

In [8]: print(simplify(e))                                                                                                        
None
```

Reply:
`Eq.simlpify` is returning `None` because of https://github.com/sympy/sympy/pull/20438#discussion_r534087321 . So wouldn't it be better to return `None` in indeterminate case ?
`None` can be changed to return `False` but that may not be correct.